app/src/main/java/org/connectbot/util/TerminalFontCompose.kt

```diff
@@ -0,0 +1,234 @@
+    storedValue: String?,
+    fallback: Typeface = Typeface.MONOSPACE
+): TerminalTypefaceResult {
+    val context = LocalContext.current
+    val fontProvider = remember { TerminalFontProvider(context) }
+    val localFontProvider = remember { LocalFontProvider(context) }
+    var typeface by remember(storedValue) { mutableStateOf(fallback) }
+    var isLoading by remember(storedValue) { mutableStateOf(true) }
+    var loadFailed by remember(storedValue) { mutableStateOf(false) }
+
+    val requestedFontName = remember(storedValue) {
+        TerminalFont.getDisplayName(storedValue)
+    }
+
+    LaunchedEffect(storedValue) {
+        Log.d(TAG, "Loading font for storedValue: $storedValue")
+        isLoading = true
+        loadFailed = false
+
+        // Check if it's a local font
+        if (LocalFontProvider.isLocalFont(storedValue)) {
+            val fileName = LocalFontProvider.getLocalFontFileName(storedValue)
+            Log.d(TAG, "Local font detected, fileName: $fileName")
+            if (fileName != null) {
+                val loadedTypeface = localFontProvider.getTypeface(fileName, fallback)
+                typeface = loadedTypeface
+                loadFailed = loadedTypeface == fallback
+            } else {
+                typeface = fallback
+                loadFailed = true
+            }
+            isLoading = false
+            return@LaunchedEffect
+        }
+
+        // Handle Google Fonts (preset or custom)
+        val googleFontName = TerminalFont.getGoogleFontName(storedValue)
+        Log.d(TAG, "Google font name resolved: '$googleFontName'")
+        if (googleFontName.isBlank()) {
+            Log.d(TAG, "Google font name is blank, using fallback")
+            typeface = fallback
+            isLoading = false
+            // Not a failure if it's SYSTEM_DEFAULT
+            loadFailed = storedValue != null &&
+                         storedValue != TerminalFont.SYSTEM_DEFAULT.name &&
+                         storedValue != TerminalFont.SYSTEM_DEFAULT.displayName
+            return@LaunchedEffect
+        }
+
+        // Check cache first
+        val cached = fontProvider.getCachedTypefaceByName(googleFontName)
+        if (cached != null) {
+            Log.d(TAG, "Font found in cache: $googleFontName")
+            typeface = cached
+            isLoading = false
+            loadFailed = false
+            return@LaunchedEffect
+        }
+
+        // Load font using suspendCancellableCoroutine for proper coroutine integration
+        Log.d(TAG, "Loading font from provider: $googleFontName")
+        val loadedTypeface = suspendCancellableCoroutine { continuation ->
+            fontProvider.loadFontByName(googleFontName) { result ->
+                if (continuation.isActive) {
+                    Log.d(TAG, "Font loaded: $googleFontName, typeface: $result")
+                    continuation.resume(result)
+                }
+            }
+        }
+        typeface = loadedTypeface
```

Comment:
I'd like this to use the `CustomFallbackBuilder` as I mentioned in another comment.